src/ui/components/components.rs

```diff
@@ -118,12 +117,23 @@ impl FriendList {
             let chat_stack = self.chat_stack.clone();
             let chat = self.chat.clone();
             let user_id = user_id.clone();
+
             move |_| {
+                let user_id = user_id.clone();
                 let mut chat = (*chat).borrow_mut();
                 chat_stack.set_visible_child(&chat.chat_element);
-                // I have no clue why I got to copy all of this but Im too tiered of fighting with
-                // the compiler at this point.
-                chat.open_chat(username.clone(), icon_path.clone(), user_id.clone());
+
+                let (tx, rx) = oneshot::channel();
+                runtime().spawn(async move {
+                    let mut headers = HashMap::new();
+                    headers.insert("Authorization", get_tokens().unwrap().discord_token.unwrap());
+
+                    let messages = ApiEndpoints::GetChannels(Some(user_id)).call(headers).await.unwrap();
```

Comment:
Why does `GetChannels` that returns channel is saved as with a variable named messages?